homeassistant/components/renault/services.py

```diff
@@ -0,0 +1,194 @@
+        vol.Optional("wednesday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("thursday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("friday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("saturday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("sunday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+    }
+)
+SERVICE_CHARGE_SET_SCHEDULES_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+        vol.Required(ATTR_SCHEDULES): vol.All(
+            cv.ensure_list, [SERVICE_CHARGE_SET_SCHEDULE_SCHEMA]
+        ),
+    }
+)
+SERVICE_CHARGE_START_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+    }
+)
+
+SERVICE_AC_CANCEL = "ac_cancel"
+SERVICE_AC_START = "ac_start"
+SERVICE_CHARGE_SET_MODE = "charge_set_mode"
+SERVICE_CHARGE_SET_SCHEDULES = "charge_set_schedules"
+SERVICE_CHARGE_START = "charge_start"
+SERVICES = [
+    SERVICE_AC_CANCEL,
+    SERVICE_AC_START,
+    SERVICE_CHARGE_SET_MODE,
+    SERVICE_CHARGE_SET_SCHEDULES,
+    SERVICE_CHARGE_START,
+]
+
+
+def setup_services(hass: HomeAssistant) -> None:
+    """Register the Renault services."""
+
+    async def ac_cancel(service_call: ServiceCall) -> None:
+        """Cancel A/C."""
+        proxy = get_vehicle_proxy(service_call.data)
+
+        LOGGER.debug("A/C cancel attempt")
+        result = await proxy.vehicle.set_ac_stop()
+        LOGGER.info("A/C cancel result: %s", result)
+
+    async def ac_start(service_call: ServiceCall) -> None:
+        """Start A/C."""
+        temperature: float = service_call.data[ATTR_TEMPERATURE]
+        when: datetime | None = service_call.data.get(ATTR_WHEN)
+        proxy = get_vehicle_proxy(service_call.data)
+
+        LOGGER.debug("A/C start attempt: %s / %s", temperature, when)
+        result = await proxy.vehicle.set_ac_start(temperature, when)
+        LOGGER.info("A/C start result: %s", result.raw_data)
+
+    async def charge_set_mode(service_call: ServiceCall) -> None:
+        """Set charge mode."""
+        charge_mode: str = service_call.data[ATTR_CHARGE_MODE]
+        proxy = get_vehicle_proxy(service_call.data)
+
+        LOGGER.debug("Charge set mode attempt: %s", charge_mode)
+        result = await proxy.vehicle.set_charge_mode(charge_mode)
+        LOGGER.info("Charge set mode result: %s", result)
+
+    async def charge_set_schedules(service_call: ServiceCall) -> None:
+        """Set charge schedules."""
+        schedules: list[dict[str, Any]] = service_call.data[ATTR_SCHEDULES]
+        proxy = get_vehicle_proxy(service_call.data)
+        charge_schedules = await proxy.vehicle.get_charging_settings()
+        for schedule in schedules:
+            charge_schedules.update(schedule)
+
+        assert charge_schedules.schedules is not None
```

Comment:
Is this assertion made for type annotating purpose?

Reply:
Yes it is, should I add TYPE_CHECKING condition?